File: public/docs/_examples/toh-4/ts/app/hero-detail.component.ts
// #docregion
import { Component, Input } from '@angular/core';
import { Hero } from './hero';

@Component({
  selector: 'my-hero-detail',
  template: `
    <div *ngIf="hero">
      <h3>{{hero.name}} details!</h3>
      <dl>
        <dt>id:</dt>
        <dd>{{hero.id}}</dd>
      </dl>
      <div>
        <label>name: 
          <input [(ngModel)]="hero.name" placeholder="name">
        </label>
      </div>
    </div>
  `
})
export class HeroDetailComponent {
  @Input() hero: Hero;
}

File: public/docs/_examples/toh-4/dart/lib/hero_detail_component.dart
// #docregion
import 'package:angular2/core.dart';

import 'hero.dart';

@Component(
    selector: 'my-hero-detail',
    template: '''
      <div *ngIf="hero != null">
        <h2>{{hero.name}} details!</h2>
        <div><label>id: </label>{{hero.id}}</div>
        <div>
          <label>name: </label>
          <input [(ngModel)]="hero.name" placeholder="name"/>
        </div>
      </div>
    ''')
class HeroDetailComponent {
  @Input()
  Hero hero;
}
